internal/services/distribution_account_service.go

```diff
@@ -0,0 +1,107 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/stellar/go/clients/horizonclient"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/data"
+	"github.com/stellar/stellar-disbursement-platform-backend/pkg/schema"
+)
+
+//go:generate mockery --name=DistributionAccountServiceInterface --case=underscore --structname=MockDistributionAccountService --filename=distribution_account_service.go
+type DistributionAccountServiceInterface interface {
+	GetBalances(ctx context.Context, account *schema.DistributionAccount) (map[string]float64, error)
+	GetBalance(ctx context.Context, account *schema.DistributionAccount, asset data.Asset) (float64, error)
+}
```

Comment:
Why is GetBalances representing an asset as a string while GetBalance represents it as a data.Asset object?

We should use data.Asset on both, so we can take advantage of its methods for comparison.